gpflow/utilities/utilities.py

```diff
@@ -382,12 +343,9 @@ def freeze(input_module: M) -> M:
     :param input_module: tf.Module or gpflow.Module.
     :return: Returns a frozen deepcopy of an input object.
     """
-    module_copy = deepcopy(input_module)
-    for name, value in parameter_dict(module_copy).items():
-        tensor = tf.convert_to_tensor(value, dtype=value.dtype)
-        const_value = tf.constant(tensor)
-        attr_path = name.lstrip(".")
-        setattr_by_path(module_copy, attr_path, const_value)
+    objects_to_freeze = _get_leaf_components(input_module)
```

Comment:
```suggestion
    objects_to_freeze = leaf_components(input_module)
```
why are there two ways of calling it, anyways?

Reply:
I prefer to leave this call to a private module method. This method was never supposed to be exposed. For some reason it was. Let's discuss it outside of this PR.